jobserver/authorization/roles.py

```diff
@@ -0,0 +1,131 @@
+from .permissions import (
+    cancel_job,
+    check_output,
+    invite_project_members,
+    publish_output,
+    review_project,
+    run_job,
+)
+
+
+class CoreDeveloper:
+    """
+    Internal user who develops and deploys opensafely core framework code.
+    """
+
+    permissions = [
+        cancel_job,
+        run_job,
+    ]
+
+
+class DataInvestigator:
+    """
+    Has access to Level 2 pseudonymised data, typically over a SQL browser, to
+    understand and debug raw underlying data. Sometimes a core developer, and
+    in rare and exceptional cases, a study developer.  Typically DataLab staff
+    only.
+    """
+
+    permissions = [
+        "",
+    ]
+
+
+class GovernanceReviewer:
+    """
+    IG, COPI coverage, and ethics for each project by each team, currently this
+    is performed by NHSE.
+    """
+
+    permissions = [
+        "",
+    ]
+
+
+class OnboardingAgent:
+    """
+    A developer and/or developer-researcher from the OpenSAFELY team with deep
+    knowledge of OpenSAFELY who is providing ongoing collaborative technical,
+    data science, analysis and academic support to external users who are
+    delivering a project, to a degree that has been pre-planned and resourced.
+    """
+
+    permissions = [
+        "",
+    ]
+
+
+class OutputChecker:
+    """
+    Review output folders that have been proposed for release, approve and
+    publish.
+    """
+
+    permissions = [
+        check_output,
+        publish_output,
```

Comment:
Based on how we've named the roles, I'd expect this to have only the permission `check_output`.   If the same user were to be able to publish *and* check outputs, then I'd expect them to have the two roles...?

Reply:
The description suggested it should be able to do both, do you want to leave it as is or update permissions and description?

Reply:
So, thinking aloud, I don't think it makes sense for separate roles for output checking and output publication when the default permissions are as they are. 

I also don't think we know yet if the checker and the publisher are always going to be the same people or not. However we do know we want two independent output checks so it's feasible that we'll want to prevent *some* checkers from also doing the releases.

It's also feasible that releases will become automatic state transitions following the acquisition of a certain number of approvals.

Overall I think the problem is that the current wording "review output folders that have been proposed for release, approve and publish" is ambiguous: I'm going to guess the writer (could have been me!) meant "review output folders (that have been proposed for release)" rather than "review output folders (that have been proposed) for release".

So I think the permissions should be non-overlapping and the description should be "Review contents of output folders that have been proposed for public release" or similar.

However this is all up for debate so let me know if you think this is still unclear

Reply:
Based on our call:

* Add a SuperUserRole
* Comment that it is expected to go away, please don't use it further
* Comment that it exists as a transition so we can assign the correct roles to the relevant staff members